recipe/meta.yaml

```diff
@@ -1,34 +1,79 @@
+# Notes: this receipt is based on torchvision-feedstock: https://github.com/conda-forge/torchvision-feedstock/blob/master/recipe/meta.yaml
+
 {% set name = "torchani" %}
-{% set version = "2.2" %}
+{% set version = "2.2.1" %}
+{% set torchani_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: c086fe6cabe49a0408d1f1b12aaf14e33b9e6ef0ffb15f3020792934e7c9fbbe
-  
+  git_url: https://github.com/aiqm/{{ name }}.git
+  git_rev: {{ version }}
+
 build:
-  noarch: python
   number: 0
-  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
+  string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}                                                  # [cuda_compiler_version == "None"]
+  string: py{{ CONDA_PY }}cuda{{ cuda_compiler_version|replace('.', '') }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}  # [cuda_compiler_version != "None"]
+  script:
+    # cuda 10.2 does not have cub, which need to be manually downloaded
+    # - sudo yum install wget unzip -y                                                  # [cuda_compiler_version == '10.2']
+    # However as the current master version of cub has conflict with cuda 10.2, cuaev will not support cuda 10.2 for now.
+    - "{{ PYTHON }} -m pip install . -vvv --no-deps"                                    # [cuda_compiler_version in ("None", '10.2')]
+    - "{{ PYTHON }} -m pip install . --global-option='--cuaev-all-sms' -vvv --no-deps"  # [cuda_compiler_version not in ("None", '10.2')]
 
 requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+    - sysroot_linux-64 ==2.17                # [linux64]
+
   host:
     - python
     - pip
     - setuptools_scm
+    - importlib_metadata
+    - numpy
+    - pytorch
+    - pytorch-gpu {{ pytorch_version }}  # [cuda_compiler_version != "None"]
+    - pytorch-cpu {{ pytorch_version }}  # [cuda_compiler_version == "None"]
+
   run:
     - python
-    - setuptools  # required for pkg_resources
-    - pytorch
+    - pip
+    - importlib_metadata
     - lark-parser
     - requests
+    - h5py
+    - numpy
+    - torchvision >=0.10
+    - pytorch
+    - pytorch-gpu {{ pytorch_version }}  # [cuda_compiler_version != "None"]
+    - pytorch-cpu {{ pytorch_version }}  # [cuda_compiler_version == "None"]
```

Comment:
`openmm-torch` seems to pin `pytorch` at https://github.com/conda-forge/openmm-torch-feedstock/blob/master/recipe/meta.yaml#L43

Need feedback here again.

Reply:
The run exports has been added in recent pytorch builds since build 3 of 1.8.0. If you drop support for 1.8.0 then you can be confident it will have it.
https://github.com/conda-forge/pytorch-cpu-feedstock/pull/43

Reply:
fixed

Reply:
@yueyericardo: use `openmm-pytorch` to pin pytorch in `run` given the last comment from @hmaarrfk

Reply:
```
    - {{ pin_compatible('pytorch-gpu', max_pin='x.x.x') }}  # [cuda_compiler_version != "None"]
    - {{ pin_compatible('pytorch-cpu', max_pin='x.x.x') }}  # [cuda_compiler_version == "None"]
```
I added this, is it the correct way to pin?

Reply:
I think so. @hmaarrfk can you confirm please?